### PR TITLE
#167727307 Email verification

### PR DESCRIPTION
**Pull request by steelze**
**What does this PR do?**
User receives an email when their account has been created by the admin

**Description of Task to be completed?**
- Generate a token and send an email verification link to user
- Verify user account
- Add swagger documentation

**Any background context you want to provide?**
When an admin creates a profile for staff, the staff should receive a verification email to verify their account.

**How should this be manually tested?**
- Fetch branch `ft-verify-account-167727307 `
- `npm install`
- `npm run migration`
- `npm run db:seed`
- `npm run start-dev`
- Create a user and copy the token received in the mail 
- Send a post request to `api\users\verify` with the token field

**What are the relevant pivotal tracker stories?**
(#167727307)[https://www.pivotaltracker.com/story/show/167727307]

### CODE REVIEW

File: src/services/userServices.js [hunk 1]
```diff
@@ -16,11 +16,11 @@ export const createSingle = async (data, res) => {
     const user = await User.create({
       username, email, firstName, lastName, roleId
     });
-    delete user.password;
+    delete user.dataValues.password;
```
Comment (Hector101) [inline src/services/userServices.js]: make use of the `omit` method of lodash.
https://lodash.com/docs/#omit

File: src/test/index.test.js [hunk 2]
```diff
@@ -39,18 +39,19 @@ describe('DEFAULT TESTS', () => {
...
+    //     .patch(`${userUrl}/roles/4`)
+    //     .send({ email: 'shonubijerry@nomad.com' })
+    // eslint-disable-next-line max-len
+    //     .set('Authorization', 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJrZXkiOnsiaWQiOjEsInJvbGVJZCI6MX0sImlhdCI6MTU2NzE4NzMwNSwiZXhwIjoxNTY3MTg3MzA4fQ.TXFOAeu71-nOk_7SYG1IFfUdaw2EJeNT-zjScuXSYro')
+    //     .end((error, res) => {
+    //       expect(res).to.have.status(401);
+    //       expect(res.body.success).to.equal(false);
+    //       expect(res.body.message).to.equal('Session has expired. Signin to continue');
+    //       done();
+    //     });
```
Comment (joaquinto) [inline src/test/index.test.js]: This looks good to me but can you remove all commented lines of codes .

File: src/controllers/userController.js [hunk 3]
```diff
@@ -102,3 +105,30 @@ export const editUserRole = async (req, res) => {
...
+ * @param {object} req
+ * @param {object} res
+ * @returns {object} response object
+ */
+export const verifyUser = async (req, res) => {
+  try {
+    const { id, email } = req.user;
+    const isUserVerified = await findSingleUser({ id, isVerified: true });
+    if (isUserVerified) {
+      return respondWithWarning(res, 400, 'User already verified');
```
Comment (Hector101) [inline src/controllers/userController.js]: make use of the `helpers/statusCode.js` to replace the status code

**Review by joaquinto — COMMENTED** [review]
LGTM

**Review by orley12 — COMMENTED** [review]
LGTM

**Pull request merged**